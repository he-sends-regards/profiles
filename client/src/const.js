export const AppRoute = {
  ROOT: '/',
  AUTH: '/auth',
  ACCOUNT: '/account',
};

export const APIRoute = {
  AUTH: '/api/auth',
  GET_USERS: '/api/users',
  DASHBOARD: '/api/users/dashboard',
  DELETE_USER: '/api/users/delete',
  GET_PROFILES: '/api/profiles/',
};

export const AuthorizationStatus = {
  AUTH: 'AUTH',
  NO_AUTH: 'NO_AUTH',
};

export const HTTPStatus = {
  OK: 200,
};

export const ProfileFormType = {
  CREATE: 'CREATE',
  EDIT: 'EDIT',
};
