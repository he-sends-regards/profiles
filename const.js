const PORT = 5000;
const jwtSecret = 'profiles1testtask1secret1string1';
const mongoUri = 'mongodb+srv://postscriptumno:HelloFrom2021@cluster0.zlo8z.mongodb.net/db?retryWrites=true&w=majority';

module.exports = {PORT, jwtSecret, mongoUri};
