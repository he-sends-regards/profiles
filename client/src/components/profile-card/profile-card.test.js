import React from 'react';
import renderer from 'react-test-renderer';
import {MenuItem} from '../../const.js';
import ProfileCard from './profile-card.jsx';

const profile = {
  _id: '1',
  owner: 'hello@gmail.com',
  birthdate: '2021-02-13',
  gender: 'Male',
  name: 'Harry Potter',
  city: 'London',
};

const noop = () => {};

it('Renders correctly', () => {
  const tree = renderer
      .create(
          <ProfileCard
            profile={profile}
            index={1}
            setIsProfileDataChanged={noop}
            setIsCardCreating={noop}
            isCardCreating={false}
            listType={MenuItem.MY_PROFILES}
          />,
      )
      .toJSON();
  expect(tree).toMatchSnapshot();
});
