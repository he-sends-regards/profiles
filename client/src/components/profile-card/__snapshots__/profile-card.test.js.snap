// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="profile-card card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Profile #
      1
    </div>
    <div
      className="mb-2 text-muted card-subtitle h6"
    >
      Harry Potter
    </div>
    <p
      className="d-flex flex-column card-text"
    >
      <span>
        Gender: 
        Male
      </span>
      <span>
        Birthdate: 
        2021-02-13
      </span>
      <span>
        City: 
        London
      </span>
    </p>
    <a
      className="card-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Edit
    </a>
    <a
      className="card-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Delete
    </a>
  </div>
</div>
`;
